src/app/model/ChessEngine.ts

```diff
@@ -67,4 +72,49 @@ export class ChessEngine implements ChessModel {
   private changeTurn(side: Side): Side {
     return side === Side.WHITE ? Side.BLACK : Side.WHITE;
   }
+
+  private willBeKingChecked(player: Player, chessboard: Chessboard, squareFrom: Square, squareTo: Square): boolean {
+    let isCheckedSquareFlag = false;
+
+    const proposedChessboard = _.cloneDeep(chessboard);
+    const { squaresWithPiece: proposedSquaresWithPieces } = proposedChessboard;
+
+    const movedPiece = chessboard.onPositionPiece(squareFrom);
+    const positionFrom = `${squareFrom.column}${squareFrom.row}`;
+    const positionTo = `${squareTo.column}${squareTo.row}`;
+    delete proposedSquaresWithPieces[positionFrom];
+    proposedSquaresWithPieces[positionTo] = movedPiece as Piece;
+
+    const allyKingPositionKey = Object.keys(proposedSquaresWithPieces).find((key) => {
```

Comment:
Tutaj też dobrze zrobić metodę np. `private kingPosition(side: Side): King`. Dobra praktyka jest, zeby metoda miała max 10 linii, takjest za trudna do zrozumienia. Duzo sie tutaj dzieje. Dobrze to podzielic na mniejsze czesci :)

Reply:
ok, dzięki za radę, zobaczę co mogę zrobić!